docs/virtual_machines/tekton_tasks.md

```diff
@@ -0,0 +1,87 @@
+- [CDI](https://github.com/kubevirt/containerized-data-importer)
+
+### What is TTO?
+[TTO](https://github.com/kubevirt/tekton-tasks-operator) is a Golang based operator, which takes care of deploying 
+[kubevirt-tekton-tasks](https://github.com/kubevirt/kubevirt-tekton-tasks) and example pipelines.
+
+### Deploying TTO
+TTO is shipped as a part of [hyperconverged-cluster-operator](https://github.com/kubevirt/hyperconverged-cluster-operator) 
+or it can be deployed by the user as a stand-alone from the latest [release](https://github.com/kubevirt/tekton-tasks-operator/releases/latest).
+
+!!! Note
+    TTO requires [Tekton](https://tekton.dev/) to work.
+
+!!! Note
+    TTO does not deploy its resources by default.
+
+The user has to enable `deployTektonTaskResources` feature gate in TTO CR or HCO CR to deploy all its resources
+
+```console
+apiVersion: tektontasks.kubevirt.io/v1alpha1
+kind: TektonTasks
+metadata:
+  name: tektontasks
+  namespace: kubevirt
+spec:
+  featureGates:
+    deployTektonTaskResources: true
+```
+
+Once `spec.featureGates.deployTektonTaskResources` is set to `true`, TTO will not delete any cluster 
+tasks or pipeline examples even if it is reverted back to false.
+
+The user can set in which namespace example pipelines will be deployed by setting `spec.pipelines.namespace`:
+```console
+apiVersion: tektontasks.kubevirt.io/v1alpha1
+kind: TektonTasks
+metadata:
+  name: tektontasks
+  namespace: kubevirt
+spec:
+  pipelines:
+    namespace: userNamespace
+```
+
+## KubeVirt tekton tasks 
+### What are Kubevirt tekton tasks?
+KubeVirt-specific Tekton tasks, which are focused on:
+
+- Creating and managing resources (VMs, DataVolumes)
+- Executing commands in VMs
+- Manipulating disk images with libguestfs tools
+
+### Existing tasks
+
+- Create Virtual Machines
+  - create-vm-from-manifest - create a VM from provided manifest
+  - create-vm-from-template - create a VM from template (works only on OpenShift)
+
+- Utilize Templates
+  - copy-template - Copies the given template and creates a new one (works only on OpenShift).
+  - modify-vm-template - Modifies a template with user provided data (works only on OpenShift).
+
+- Create DataVolumes
+  - create-datavolume-from-manifest - Create a datavolume from a manifest.
+
+- Generate SSH Keys
+  - generate-ssh-keys - Generates a private and public key pair, and injects it into a VM.
+
+- Execute commands in Virtual Machines
+  - execute-in-vm: Execute commands over SSH in a VM.
+  - cleanup-vm: Execute commands and/or stop/delete VMs.
+
+- Manipulate PVCs with libguestfs tools
+  - disk-virt-customize: execute virt-customize commands in PVCs.
+  - disk-virt-sysprep: execute virt-sysprep commands in PVCs.
+
+- Wait for Virtual Machine Instance Status
+  - wait-for-vmi-status - Waits for a VMI to be running.
+
+All these tasks can be used for creating [pipelines](https://github.com/kubevirt/tekton-tasks-operator/tree/main/data/tekton-pipelines/okd) which can work with VMs.
```

Comment:
If these are examples of pipelines created using the features you describe here, it would be clearer to call them out as examples, perhaps even making it a heading to gain attention. As it stands, the link text implies that it would just show the reader what pipelines are in general.